rustfmt.toml

```diff
@@ -0,0 +1,5 @@
+# Defaults are good.  Let's keep the defaults except if we really, really,
```

Comment:
👍

Reply:
Does this comment mean we should just not have a custom rustfmt.toml?

Also, are these settings we enabled all available on stable?

Reply:
Might not be as the rustfmt job failed

Reply:
![image](https://user-images.githubusercontent.com/552961/91911627-78ce2900-ec7f-11ea-82a6-4260167742c4.png)

Reply:
No, neither of them are stable.  For the comment, I just meant that we should try to use the defaults unless we have some good reason to do something else (like import grouping), rather than going through every rustfmt option and spending a lot of time on deciding whether we should use each one or not.

Reply:
yea, I don't think we should use nightly rustfmt settings personally, this will make contributing to our project a lot more annoying because people will need to either use nightly or configure their dev env to use nightly cargo for fmting regardless of the default toolchain.

Reply:
Ah, if we have to change to nightly rustfmt maybe we don't want to do this. I'm happy either way.  I think that the code is nicer with these options enabled but I don't care that much.

Reply:
OK cool, let's not do this then.

Reply:
![image](https://user-images.githubusercontent.com/552961/91911975-2b9e8700-ec80-11ea-84c0-6b6bcf1a66c3.png)

We can make the CI rustfmt use nightly if we want, as long as we all remember to use that locally...

Reply:
Yeah, I think that @yaahc's point was that this is an extra barrier to new contributors; I think that the benefit is pretty marginal compared to that cost.